src/components/ui/phone-input.tsx

```diff
@@ -35,6 +35,31 @@ type PhoneInputProps = Omit<
 const PhoneInput: React.ForwardRefExoticComponent<PhoneInputProps> =
   React.forwardRef<React.ElementRef<typeof RPNInput.default>, PhoneInputProps>(
     ({ className, onChange, ...props }, ref) => {
+      const getCountryFromNumber = (phoneNumber: string) => {
+        const parsedNumber = RPNInput.parsePhoneNumber(phoneNumber);
+        return parsedNumber?.country || careConfig.defaultCountry;
+      };
+
+      const [selectedCountry, setSelectedCountry] =
+        React.useState<RPNInput.Country>(
+          props.value
```

Comment:
modify getCountryFromNumber to return defaultCountry when value isn't present, can remove this check.